### ISSUE TITLE
Add support for Assured Workloads

### ISSUE THREAD

**Issue by rileykarson** [hashicorp/terraform-provider-google#8673]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### New or Affected Resource(s)

* google_assured_workloads_workload

### Potential Terraform Configuration

```tf
resource "google_assured_workloads_workload" "my_workload" {
  name = "my-workload"
  organization = "1234567890"
  location = "us-central1" # it's unclear what's actually expected for this value

  display_name = "My Workload"
  compliance_regime = "FEDRAMP_HIGH"
  billing_account = "billingAccounts/1234567890"

  resources {
    resource_id = "1234567890" # (project number)
    resource_type = "CONSUMER_PROJECT"
  }

  resources {
    resource_id = "0987654321"
    resource_type = "ENCRYPTION_KEYS_PROJECT"
  }
}
```

### References

* https://cloud.google.com/assured-workloads/docs/reference/rest/v1/organizations.locations.workloads
* b/181989532

**Comment by rileykarson** [comment]
note: DCL support is available, pending release

**Comment by rileykarson** [comment]
Unassigning- no DCL support yet, I'd misread the status. Stephen, I'm gonna swap this with https://github.com/hashicorp/terraform-provider-google/issues/6694 which is available.

**Comment by ScottSuarez** [comment]
closed with https://github.com/GoogleCloudPlatform/magic-modules/pull/4936

**Comment by pgoldtho** [comment]
@ScottSuarez should this be reopened since #4936 was reverted?

I'm guessing it failed in testing. I encountered a couple of issues while testing a local build. The destroy action fails because the Cloud Resource Manager API is not enabled for the workload. 

```
Error: Error deleting Workload: googleapi: Error 403: Cloud Resource Manager API has not been used in project before or it is disabled. 
```

I was able to delete the resource after manually enabling the API but it required 2 attempts. The first attempt failed with:

```
Error: Error deleting Workload: failed to delete Workload: googleapi: Error 400: Workload contains projects or other resources that are not deleted.
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.ErrorInfo",
    "domain": "assuredworkloads.googleapis.com",
    "reason": "ERROR_CODE_WORKLOAD_NOT_EMPTY"
  }
]
```

I was able to delete the workload by rerunning `terraform apply`. I'm assuming that is because projects associated with the workload had been marked for deletion in the first run. 

I suspect these are issues with the Assured Workload API rather than its terraform wrapper.

**Comment by ScottSuarez** [comment]
That revert was specifically for release `3.75` but we will have to revert it for `3.76` as well. Assured workloads api is rapidly developing and some assumptions we made broke our upstream implementation as the api has mutated.

**Comment by ScottSuarez** [comment]
It will be available in `3.76` pending no new issue.
